### DISCUSSION TITLE
Next.js app with prisma database can't connect with each other

### DISCUSSION THREAD

**Discussion by slashinteractive** [Dokploy/dokploy#992] [Q&A] [answered]
Hi Everyone, 

I have a next.js app that is deployed on Dokploy with a Dockerfile. I also have a seperate mysql database on dokploy successfully running. 

I use the "Internal Connection URL" that dokploy provides as the value of the DATABASE_URL that prisma requires. The build of the next.js app fails on the "Generating static pages" phase, because it can't connect to the database (the .env value is picked up, because I see the name of the database, but it fails to connect regardless).

I also tried to make the database available over the internet, but I have to open port 3306 on my firewall which isn't something I want.

I suspect it's because of "dokploy-network" that I have to set inside my Dockerfile, but unsure how and if that's going to help. 

I'd appreciate if anyone could provide some insight in the matter.

Thanks in advance.

**Comment by akparhi** [comment, marked as answer]
You need to expose it and use the external connection string.

**Reply by slashinteractive** [reply]
Thanks, I got it working that way, but I was wondering: what's the purpose of the internal url?

**Reply by akparhi** [reply]
Internal URL does not seem to work due to a upstream bug.

**Reply by Hoshino-Yumetsuki** [reply]
Yes, I'm also having the same issue, my postgresql doesn't work with internal URLs

**Reply by nktnet1** [reply]
It should work if you deploy using a Dockerfile or docker compose, and have added a domain.

The important part being that your NextJS container is connected to `dokploy-network` in order to access internal URLs.

You can read more about this in the thread below:
- https://github.com/Dokploy/dokploy/issues/507
